### ISSUE TITLE
Failed to connect to Chrome DevTools Protocol after 500 in Electron - Cypress 12.12.0+

### ISSUE THREAD

**Issue by estefafdez** [currents-dev/cypress-debugger#40]
Hello again, 

we follow every step on the readme and it's working properly in Chrome but not in Electron. When we run the tests in open/run mode, we got the following errors: 

Run mode:
![image](https://github.com/currents-dev/cypress-debugger/assets/6990209/613f35de-7ce3-4e43-b97e-d735d75b6b0a)

Open mode: 
<img width="1426" alt="image" src="https://github.com/currents-dev/cypress-debugger/assets/6990209/06eec4de-33a9-44e7-9e5d-97189cad01ab">
<img width="499" alt="image" src="https://github.com/currents-dev/cypress-debugger/assets/6990209/6ea75684-efd1-4711-ae4d-88664d8662c1">

`````
 at <unknown> (https://auth-xx-xx-xx.xx.ly/__cypress/runner/cypress_runner.js:150983:78)
From previous event:
    at Context.task (https://auth-xx-xx-xx.xx.ly/__cypress/runner/cypress_runner.js:150968:15)
    at wrapped (https://auth-xx-xx-xx.xx.ly/__cypress/runner/cypress_runner.js:160605:43)
From Your Spec Code:
    at Context.eval (https://auth-xx-xx-xx.xx.ly/__cypress/tests?p=cypress/support/e2e.ts:348940:21)
From Node.js Internals:
  Error: Failed to connect to Chrome Debugging Protocol
  
  Possible reasons for failure:
    - Chrome not running in headless mode
    - Using Chrome version 58 or earlier
    - Inconsistent RDP configuration settings.
    
  The stack trace for this error is:
      at V.<anonymous> (webpack://@neuralegion/cypress-har-generator/./src/cdp/CDPConnection.ts:55:15)
  at Generator.next (<anonymous>)
      at fulfilled (/xx/xx/xx/xx/qa/e2e/node_modules/tslib/tslib.js:115:62)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
`````

We usually used `yarn` to manage the project, but we tried with `npx` as well and we got the same error. 

More details: 
- Cypress version: 12.13.0 
- Electron version: v106
- Cypress-debugger version: ^1.0.3
- OS: MacOS Ventura 13.3.1 (a) and Ubuntu 22. (same problem)

**Comment by agoldis** [comment]
@estefafdez thanks for your patience.
I have submitted the troubleshooting guide here: https://github.com/currents-dev/cypress-debugger#troubleshooting
Obviously it's a bit hard to debug without access to the whole system. 

I validated that the plugin is working well with the bare setup as appears [here](https://github.com/currents-dev/cypress-debugger/tree/main/apps/web).


https://github.com/currents-dev/cypress-debugger/assets/1637928/eef91a63-4cb3-4d2b-9904-86cdc5a0b664


I know it's annoying, but it's kind of "working on my machine". Please try enabling the debug mode (as appears in the troubleshooting guide - maybe we can find more clues in the logs.

**Comment by estefafdez** [comment]
Hello again @agoldis, thanks for your message, we added the debug mode and those are our logs when running: 

`DEBUG=cypress:server:browsers:electron,cypress-debugger* ELECTRON_EXTRA_LAUNCH_ARGS=--remote-debugging-port=9222 npx cypress run --env configFile=qa --browser electron --spec='cypress/test/a_spec.ts'`

````
DevTools listening on ws://127.0.0.1:57217/devtools/browser/99e049e5-2686-4383-8b73-67b605a34e1e
Missing baseUrl in compilerOptions. tsconfig-paths will be skipped

 Running:  a_spec.ts  (1 of 1)
  cypress:server:browsers:electron open { browser: { name: 'electron', channel: 'stable', family: 'chromium', displayName: 'Electron', version: '106.0.5249.51', path: '', majorVersion: 106, isHeadless: true, isHeaded: false }, url: '*' } +0ms
  cypress:server:browsers:electron received saved state {} +1ms
  cypress:server:browsers:electron browser window options { browser: { name: 'electron', channel: 'stable', family: 'chromium', displayName: 'Electron', version: '106.0.5249.51', path: '', majorVersion: 106, isHeadless: true, isHeaded: false }, url: '*', browsers: [ { name: 'chrome', family: 'chromium', channel: 'stable', displayName: 'Chrome', version: '114.0.5735.106', path: '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome', minSupportedVersion: 64, majorVersion: '114' }, { name: 'electron', channel: 'stable', family: 'chromium', displayName: 'Electron', version: '106.0.5249.51', path: '', majorVersion: 106, isHeadless: true, isHeaded: false } ], userAgent: null, proxyUrl: 'http://localhost:57218', proxyServer: 'http://localhost:57218', socketIoRoute: '/__socket', chromeWebSecurity: false, isTextTerminal: true, downloadsFolder: '*', experimentalModifyObstructiveThirdPartyCode: false, experimentalWebKitSupport: true, projectRoot: '*', shouldLaunchNewTab: false, videoApi: undefined, automationMiddleware: { onBeforeRequest: [Function: onBeforeRequest], onAfterResponse: [Function: onAfterResponse] }, x: null, y: null, width: 1280, height: 720, minWidth: 100, minHeight: 100, devTools: false, contextMenu: true, partition: 'persist:run-71906', trackState: { width: 'browserWidth', height: 'browserHeight', x: 'browserX', y: 'browserY', devTools: 'isBrowserDevToolsOpen' }, webPreferences: { sandbox: true, partition: null, webSecurity: true, nodeIntegration: false, backgroundThrottling: false }, show: false, resizable: false, frame: true, recordFrameRate: null } +1ms
  cypress:server:browsers:electron launching browser window to url: * +2ms
  cypress:server:browsers:electron clearing cache +249ms
  cypress:server:browsers:electron debugger: enable Console and Network +1ms
  cypress-debugger Attempting to connect to Chrome DevTools Protocol +0ms
  cypress-debugger Failed to connect to Chrome DevTools Protocol, attempt: 1 +14ms
  cypress-debugger Reconnecting... +1ms
  cypress-debugger Failed to connect to Chrome DevTools Protocol, attempt: 2 +224ms
  cypress-debugger Reconnecting... +0ms
  cypress-debugger Failed to connect to Chrome DevTools Protocol, attempt: 3 +117ms
  cypress-debugger Reconnecting... +0ms
  cypress-debugger Failed to connect to Chrome DevTools Protocol, attempt: 4 +671ms
  cypress-debugger Reconnecting... +0ms
An error was thrown in your plugins file while executing the handler for the before:spec event.

The error we received was:

Error: Failed to connect to Chrome DevTools Protocol after 500
    at o (*/node_modules/@currents/cypress-debugger-plugin/src/browserLogs.ts:101:15)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async o (*/node_modules/@currents/cypress-debugger-plugin/src/browserLogs.ts:118:9)
    at async o (*node_modules/@currents/cypress-debugger-plugin/src/browserLogs.ts:118:9)
    at async o (*/node_modules/@currents/cypress-debugger-plugin/src/browserLogs.ts:118:9)
    at async o (*/node_modules/@currents/cypress-debugger-plugin/src/browserLogs.ts:118:9)
    at async J (*/node_modules/@currents/cypress-debugger-plugin/src/browserLogs.ts:123:3)
    at async w (*/node_modules/@currents/cypress-debugger-plugin/src/browserLogs.ts:155:3)
    at async Object.handler (*/node_modules/@currents/cypress-debugger-plugin/src/install.ts:93:5)
  cypress:server:browsers:electron closed event fired +3s
...
````

I removed some URLs and internal data from the log but nothing important. As it is a private project, I'm going to try to replicate the same problem on a project that I had with almost the same file/plugin structure as ours (https://github.com/estefafdez/cypress-template). If it works, you can play with it until discover the failure, if not, we can set up a meeting to have a look together at the problem and see what's going on.

**Comment by estefafdez** [comment]
Ok @agoldis , sorry for the delay, there you have a PR with some changes (exactly how we manage our private project) and with the debugger added. Same failure in electron. You can have a look and debug it :) 

https://github.com/estefafdez/cypress-template/pull/88

**Comment by agoldis** [comment]
@estefafdez 
This is happening on Cypress 12.12.0+ because of this change:
- https://github.com/cypress-io/cypress/pull/26573

And is being handled here:
- https://github.com/cypress-io/cypress/issues/26711
- https://github.com/NeuraLegion/cypress-har-generator/issues/288

**Comment by estefafdez** [comment]
Wow! That was fast! I will have a look at those issues. Thanks!!

**Comment by agoldis** [comment]
Seems like the issue is just about to be fixed: https://github.com/cypress-io/cypress/blob/e65a3e348a9ba386032263cb4af5934e28071b67/cli/CHANGELOG.md#12170

**Comment by estefafdez** [comment]
Hey @agoldis , the issue is supposed to be fixed on cypress 12.17.0 but we still are facing the same problem with the task (https://github.com/cypress-io/cypress/issues/27163) and electron. Is there any problem with the task statement on the plugin? could you please check on your side if is a cypress problem or something inside the plugin? thanks!

**Comment by agoldis** [comment]
@estefafdez the connection to electron was fixed indeed, but now we get a different error.
Seems like it is related to [#27149](https://github.com/cypress-io/cypress/issues/27149) and [#27163](https://github.com/cypress-io/cypress/issues/27163) - even though marked as fixed, the error message resemble the errors reporting. 
Investigating further...

**Comment by agoldis** [comment]
Confirmed that it's an ongoing issue with cypress internal bug @estefafdez thanks a lot for your help and escalating !

**Comment by estefafdez** [comment]
Happy to help @agoldis!

**Comment by nishant-shah-social** [comment]
@agoldis Thanks for the debugger.
Somehow, I am also getting this error when trying to integrate debugger to cypress 12.17.3
From what I understand via this conversation is this issue is related to cypress, so will wait for the cypress to fix it.
Thanks!

**Comment by agoldis** [comment]
@nishant-shah-social should have been fixed in 12.17.2, are you sure the right version is being used?

**Comment by nishant-shah-social** [comment]
@agoldis this issue got resolved for me but facing another issue with chrome now. May be will share the error with you separately and not on this ticket
Thanks!
